src/Pmi/Application/AbstractApplication.php

```diff
@@ -373,10 +373,17 @@ protected function enableTwig()
         }));
 
         // Register custom Twig path_exists function
-        $this['twig']->addFunction(new Twig_SimpleFunction('path_exists', function($name) {
+        $this['twig']->addFunction(new Twig_SimpleFunction('path_exists', function ($name) {
             return !is_null($this['routes']->get($name));
         }));
 
+        // Convert a string into a slug
+        $this['twig']->addFilter(new Twig_SimpleFilter('slugify', function ($text) {
+            $output = trim(strtolower($text));
+            $output = str_replace([' ', '-', '(', ')', ':'], '-', $output);
```

Comment:
Based on my understanding of the requirements, I'm not seeing a need to include any characters other than letters and digits.  What about something like:

```php
$output = preg_replace('/[^a-z0-9]/', '-', $output);
```

Reply:
❤️